### PR TITLE
Implement provider options

### PR DESCRIPTION
**Pull request by jacobweinstock**
## Description

This implements the provider options now available in the connection crd for machine and task controllers. Also, updated dependencies for controller-runtime and kubernetes client libraries.

## Why is this needed

Fixes: #

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -119,13 +116,15 @@ func main() {
...
-		Namespace:              kubeNamespace,
+		Scheme: scheme,
+		Metrics: metricsserver.Options{
+			BindAddress: metricsAddr,
+		},
+		// Port:                   9443,
+		HealthProbeBindAddress:  probeAddr,
+		LeaderElection:          enableLeaderElection,
+		LeaderElectionID:        "e74dec1a.tinkerbell.org",
+		LeaderElectionNamespace: kubeNamespace,
```
Comment (jacobweinstock) [inline main.go:127]: Hey @chrisdoherty4, i could use your eyes here. I upgraded controller-runtime to v0.16.2 and didn't fully know how or if i correctly translated all the options over.

Comment (chrisdoherty4) [reply main.go:127]: The old `manager.Options.Namespace` is now `manager.Options.Cache.Namespaces`. We can modify the CLI params separately to allow specifying multiple.

We don't need to configure `LeaderElectionNamepsace`.

For the `Port` field we need to build a webhook server using `webhook.NewServer`. It accepts an options struct that we can configure the port on and then set `manager.Options.WebhookServer`.

File: controller/machine.go [hunk 2]
```diff
@@ -127,99 +140,72 @@ func (r *MachineReconciler) doReconcile(ctx context.Context, bm *v1alpha1.Machin
...
 	}
 
-	return ctrl.Result{}, utilerrors.Flatten(aggErr)
+	return ctrl.Result{RequeueAfter: machineRequeueInterval}, nil
 }
 
-// reconcilePower ensures the Machine Power is in the desired state.
-func (r *MachineReconciler) reconcilePower(ctx context.Context, bm *v1alpha1.Machine, bmcClient *bmclib.Client) error {
+// powerState gets the current power state of the machine.
+func (r *MachineReconciler) powerState(ctx context.Context, bm *v1alpha1.Machine, bmcClient *bmclib.Client) error {
```
[the remark this answers is not in the document]
Comment (jacobweinstock) [reply controller/machine.go:167]: getting a power state from a machine isn't a reconciliation though, is it? whats the definition of reconcile? I dont know if i understand it in this context. My understanding is that when a a kubernetes object changes, a controller's job is to "reconcile" that change to make the "real" world reflect that change. With this definition in mind, this func does not do this. Its making the kubernetes object (status) reflect the state of the "real" world.

[the remark this answers is not in the document]
Comment (chrisdoherty4) [reply controller/machine.go:167]: I empathize, I'm ok with `reconcile` or `update`. Either way, at the consumer site reading `powerState` is less useful, I feel, than `updatePowerState` or `reconcilePowerState` which seem far clearer in the intent.

File: controller/machine.go [hunk 3]
```diff
@@ -127,99 +140,72 @@ func (r *MachineReconciler) doReconcile(ctx context.Context, bm *v1alpha1.Machin
...
 	rawState, err := bmcClient.GetPowerState(ctx)
 	if err != nil {
-		r.recorder.Eventf(bm, corev1.EventTypeWarning, EventGetPowerStateFailed, "get power state: %v", err)
+		bm.Status.Power = v1alpha1.Unknown
+		r.recorder.Eventf(bm, corev1.EventTypeWarning, eventGetPowerStateFailed, "get power state: %v", err)
```
[the remark this answers is not in the document]
Comment (jacobweinstock) [reply controller/machine.go:171]: sure thing. To which sibling are you referring?

File: main.go (hunk 1 above)
Comment (jacobweinstock) [reply main.go:127]: I'm a bit confused. Are we even deploying a webhook in Rufio?

**Review by chrisdoherty4 — APPROVED** [review]

**Comment by jacobweinstock** [comment]
@mergifyio queue

**Comment by jacobweinstock** [comment]
@mergifyio refresh

**Comment by jacobweinstock** [comment]
@mergifyio queue

**Comment by jacobweinstock** [comment]
@mergifyio refresh

**Comment by jacobweinstock** [comment]
@Mergifyio unqueue

**Comment by jacobweinstock** [comment]
@mergifyio queue

**Pull request merged**